src/overlay/OverlayManagerImpl.h

```diff
@@ -95,6 +95,34 @@ class OverlayManagerImpl : public OverlayManager
 
     std::shared_ptr<SurveyManager> mSurveyManager;
 
+    // This gets called once when starting
+    // and it continues to call itself every FLOOD_DEMAND_PERIOD_MS.
+    void demand();
+    VirtualTimer mDemandTimer;
+    struct DemandHistory
+    {
+        VirtualClock::time_point lastDemanded;
+        UnorderedSet<Peer::pointer> peers;
```

Comment:
I meant the NodeID, we use those in a bunch of places across the codebase.

Reply:
I changed it to NodeID since it seems simple & I ran into issues with creating unordered sets with weak ptrs (apparently a known issue). I didn't see much benefit in switching to a regular set with weak ptrs, but let me know if you disagree.